java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/util/KcqlConfigBusMapper.java

```diff
@@ -0,0 +1,124 @@
+  private static final Pattern BUS_TYPE_PATTERN = Pattern.compile(BUS_TYPE_REGEX);
+  private static final String TOPIC_NAME_ERROR_MESSAGE =
+      "%s topic %s, name is not correctly specified: It can contain only letters, numbers and hyphens,"
+          + " underscores and dots and has to start with number or letter with max size of " + MAX_BUS_NAME_LENGTH;
+
+  private static final String SUBSCRIPTION_NAME_ERROR_MESSAGE =
+      "Subscription name %s is not correctly specified but is necessary for TOPIC: "
+          + "It can contain only letters, numbers and hyphens, underscores and dots and "
+          + "has to start with number or letter with max size of " + MAX_SUBSCRIPTION_LENGTH;
+
+  private static final String BUS_TYPE_ERROR_MESSAGE =
+      "Property %s contains invalid value. Valid values are: TOPIC or QUEUE";
+  private static final List<ServiceBusKcqlProperties> NECESSARY_PROPERTIES =
+      ServiceBusKcqlProperties.getNecessaryProperties();
+
+  /**
+   * This method parses KCQL statements and fetches input and output topics checking against regex for invalid
+   * topic names in input and output as well as if mappings contain necessary KCQL properties.
+   * 
+   * @param kcqlString string to parse
+   * @return list of KCQLs if parsed properly
+   */
+  public static List<Kcql> mapKcqlsFromConfig(String kcqlString) {
+    List<Kcql> kcqls = Kcql.parseMultiple(kcqlString);
+    Map<String, String> inputToOutputTopics = new HashMap<>(kcqls.size());
+
+    for (Kcql kcql : kcqls) {
+      String inputTopic = kcql.getSource();
+      String outputTopic = kcql.getTarget();
+
+      if (!azureNameMatchesAgainstRegex(inputTopic, MAX_BUS_NAME_LENGTH)) {
+        throw new ConfigException(String.format(TOPIC_NAME_ERROR_MESSAGE, "Input", inputTopic));
+      }
+      if (!azureNameMatchesAgainstRegex(outputTopic, MAX_BUS_NAME_LENGTH)) {
+        throw new ConfigException(String.format(TOPIC_NAME_ERROR_MESSAGE, "Output", outputTopic));
+      }
+      if (inputToOutputTopics.containsKey(inputTopic)) {
+        throw new ConfigException(String.format("Input %s cannot be mapped twice.", inputTopic));
+      }
+      if (inputToOutputTopics.containsValue(outputTopic)) {
+        throw new ConfigException(String.format("Output %s cannot be mapped twice.", outputTopic));
+      }
+
+      List<ServiceBusKcqlProperties> notSatisfiedProperties = checkForNecessaryKcqlProperties(kcql);
+      if (!notSatisfiedProperties.isEmpty()) {
+        String missingPropertiesError =
+            notSatisfiedProperties.stream()
+                .map(ServiceBusKcqlProperties::getPropertyName)
+                .collect(Collectors.joining(","));
+        throw new ConfigException(
+            String.format("Following non-optional properties missing in KCQL: %s", missingPropertiesError));
+      }
+
+      checkForValidPropertyValues(kcql.getProperties());
+
+      inputToOutputTopics.put(inputTopic, outputTopic);
+    }
+
+    return kcqls;
+  }
```

Comment:
Could we split the validation logic and mapping logic into separate concerns?

Something like (this isn't a complete implementation):

```suggestion
  public static List<Kcql> mapKcqlsFromConfig(String kcqlString) {
    List<Kcql> kcqls = Kcql.parseMultiple(kcqlString);

    Map<String, String> inputToOutputTopics = kcqls.stream()
            .collect(Collectors.toMap(Kcql::getSource, Kcql::getTarget));

    validateMappings(inputToOutputTopics);

    return kcqls;
  }
```

Reply:
I splitted the mapping and validation logic to make it more clear (thanks for all advices here!)

Reply:
Done (i)